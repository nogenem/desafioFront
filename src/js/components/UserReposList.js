const renderUserRepoInfo = ({
  full_name,
  name,
  stargazers_count,
  forks_count,
  open_issues
}) => `
  <tr data-fullname="${full_name}">
    <td>${name}</td>
    <td>${stargazers_count}</td>
    <td>${forks_count}</td>
    <td>${open_issues}</td>
  </tr>
`;

const renderUserReposList = repos => `
  <table id="reposTable" class="display" cellspacing="0" width="100%">
    <thead>
      <tr>
        <th>Nome</th>
        <th>Stars</th>
        <th>Forks</th>
        <th>Problemas Abertos</th>
      </tr>
    </thead>
    <tbody>
      ${repos.map(renderUserRepoInfo).join("")}
    </tbody>
  </table>
`;

export default {
  render: renderUserReposList,
  didMount: () => {
    $("#reposTable").DataTable({
      order: [[1, "desc"]]
    });
    document.querySelector("#reposTable tbody").addEventListener(
      "click",
      e => {
        e.preventDefault();

        const { fullname, user } = e.srcElement.parentElement.dataset;
        location.hash = `#/user/${fullname}`;
      },
      false
    );
  }
};
