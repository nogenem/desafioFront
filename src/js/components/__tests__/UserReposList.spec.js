import UserReposList from "../UserReposList";

const testData = [
  {
    full_name: "test/testProj",
    name: "testProj",
    stargazers_count: "0",
    forks_count: "0",
    open_issues: "0"
  }
];

it("renders correctly", () => {
  const output = UserReposList.render(testData);
  expect(output).toMatchSnapshot();
});
