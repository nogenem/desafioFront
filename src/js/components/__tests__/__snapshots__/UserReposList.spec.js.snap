// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
"
  <table id=\\"reposTable\\" class=\\"display\\" cellspacing=\\"0\\" width=\\"100%\\">
    <thead>
      <tr>
        <th>Nome</th>
        <th>Stars</th>
        <th>Forks</th>
        <th>Problemas Abertos</th>
      </tr>
    </thead>
    <tbody>
      
  <tr data-fullname=\\"test/testProj\\">
    <td>testProj</td>
    <td>0</td>
    <td>0</td>
    <td>0</td>
  </tr>

    </tbody>
  </table>
"
`;
